src/toolkits/object_detection/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -209,23 +210,28 @@ gl_sframe augment_data(const gl_sframe &data,
    * Replacing the `for` with a `parallel_for` fails the export_coreml unit test
    * with an EXC_BAD_ACCESS in the function call to boost::gil::resample_pixels
    */
-  for (size_t segment_id = 0; segment_id < nsegments; segment_id++) {
-    size_t segment_start = (segment_id * backgrounds.size()) / nsegments;
-    size_t segment_end = ((segment_id + 1) * backgrounds.size()) / nsegments;
-    size_t row_number = segment_start;
-    for (const auto &background_ft :
-         backgrounds.range_iterator(segment_start, segment_end)) {
-      row_number++;
-      flex_image flex_background =
-          image_util::decode_image(background_ft.to<flex_image>());
-      for (const auto &row : decompressed_data.range_iterator()) {
-        // go through all the starter images and create augmented images for
-        // all starter images and the respective chunk of background images
-        const flex_image &object = row[image_column_index].get<flex_image>();
-        std::string label = row[target_column_index].to<flex_string>();
+  for (const auto &row : decompressed_data.range_iterator()) {
```

Comment:
Can you avoid using `auto` here and on line 225?

Reply:
FYI: https://google.github.io/styleguide/cppguide.html#Type_deduction

> The fundamental rule is: use type deduction only to make the code clearer or safer, and do not use it merely to avoid the inconvenience of writing an explicit type.